### PR TITLE
[IOTDB-3800]Add Node Type Column to 'SHOW CHILD PATHS' query

### PR DESCRIPTION
**Pull request by yifuzhou**
## Description
image
image
image
 
All look well.

### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/IMTreeBelowSG.java [hunk 1]
```diff
@@ -136,8 +137,8 @@ List<MeasurementPath> getMeasurementPaths(PartialPath pathPattern, boolean isPre
    *
    * @param pathPattern a path pattern or a full path, may contain wildcard
    * @param isPrefixMatch if true, the path pattern is used to match prefix path
-   * @return Pair.left contains all the satisfied paths Pair.right means the current offset or zero
-   *     if we don't set offset.
+   * @return Pair.left contains all the satisfied getChildNodePathInNextLevel Pair.right means the
+   *     current offset or zero if we don't set offset.
    */
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/mtree/IMTreeBelowSG.java:142]: Changes on this comment seems not appropriate.

File: server/src/main/java/org/apache/iotdb/db/metadata/mnode/IMNode.java [hunk 2]
```diff
@@ -88,6 +88,22 @@ public interface IMNode extends Serializable {
...
+      return MNodeType.STORAGE_GROUP;
+    }
+    if (isMeasurement()) {
+      return MNodeType.MEASUREMENT;
+    }
+    if (isConfig) {
+      return MNodeType.SG_INTERNAL;
+    }
+    return MNodeType.INTERNAL;
+  }
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/mnode/IMNode.java:105]: It's better to push down these if-else judgement to specific MNode class with method override.

**Review by MarcosZyk — COMMENTED** [review]
Excellent work! Only a few problems. PTAL.

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/IMTreeBelowSG.java (hunk 1 above)
Comment (yifuzhou) [reply server/src/main/java/org/apache/iotdb/db/metadata/mtree/IMTreeBelowSG.java]: reverted it.

File: server/src/main/java/org/apache/iotdb/db/metadata/mnode/IMNode.java (hunk 2 above)
Comment (yifuzhou) [reply server/src/main/java/org/apache/iotdb/db/metadata/mnode/IMNode.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/metadata/mnode/StorageGroupEntityMNode.java [hunk 3]
```diff
@@ -81,6 +81,11 @@ public boolean isStorageGroup() {
     return true;
   }
 
+  @Override
+  public MNodeType getMNodeType(Boolean isConfig) {
+    return MNodeType.STORAGE_GROUP;
+  }
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/mnode/StorageGroupEntityMNode.java:87]: It seems the ```StorageGroupEntityMNode``` is only described as storage group? What if the user want to check whether it is device? Or it's just a rule and we don't suggest user to reach into such case?

**Comment by MarcosZyk** [comment]
Please update the User-Guide doc.

File: server/src/main/java/org/apache/iotdb/db/metadata/mnode/StorageGroupEntityMNode.java (hunk 3 above)
Comment (yifuzhou) [reply server/src/main/java/org/apache/iotdb/db/metadata/mnode/StorageGroupEntityMNode.java:87]: Yes, in current mpp node management framework, it is a little bit hard to define one node both as sg and device. If user want to know whether it is device, maybe need to try another sql like show child paths for this path to see whether has measurement child node. I will imporve it from the next pr.

**Review by MarcosZyk — APPROVED** [review]
LGTM~

**Review by SpriCoder — COMMENTED** [review]
LGTM~

**Pull request merged**